aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -255,3 +363,151 @@ func flattenProductionVariants(list []*sagemaker.ProductionVariant) []map[string
+		"capture_options": flattenSagemakerCaptureOptions(dataCaptureConfig.CaptureOptions),
+		//"capture_content_type_header": flattenSagemakerCaptureContentTypeHeader(dataCaptureConfig.CaptureContentTypeHeader),
+	}
+
+	if dataCaptureConfig.EnableCapture != nil {
+		cfg["enable_capture"] = aws.BoolValue(dataCaptureConfig.EnableCapture)
+	}
+
+	if dataCaptureConfig.KmsKeyId != nil {
+		cfg["kms_key_id"] = aws.StringValue(dataCaptureConfig.KmsKeyId)
+	}
+
+	if dataCaptureConfig.CaptureContentTypeHeader != nil {
+		cfg["capture_content_type_header"] = flattenSagemakerCaptureContentTypeHeader(dataCaptureConfig.CaptureContentTypeHeader)
+	}
+
+	return []map[string]interface{}{cfg}
+}
+
+func expandSagemakerCaptureOptions(configured []interface{}) []*sagemaker.CaptureOption {
+	containers := make([]*sagemaker.CaptureOption, 0, len(configured))
+
+	for _, lRaw := range configured {
+		data := lRaw.(map[string]interface{})
+
+		l := &sagemaker.CaptureOption{
+			CaptureMode: aws.String(data["capture_mode"].(string)),
+		}
+		containers = append(containers, l)
+	}
+
+	return containers
+}
+
+func flattenSagemakerCaptureOptions(list []*sagemaker.CaptureOption) []map[string]interface{} {
+	containers := make([]map[string]interface{}, 0, len(list))
+
+	for _, lRaw := range list {
+		captureOption := make(map[string]interface{})
+		captureOption["capture_mode"] = aws.StringValue(lRaw.CaptureMode)
+		containers = append(containers, captureOption)
+	}
+
+	return containers
+}
+
+func expandSagemakerCaptureContentTypeHeader(m map[string]interface{}) *sagemaker.CaptureContentTypeHeader {
+	c := &sagemaker.CaptureContentTypeHeader{}
+
+	if v, ok := m["csv_content_types"]; ok && len(v.([]interface{})) > 0 {
+		contentTypes := make([]*string, 0, len(v.([]interface{})))
+
+		for _, raw := range v.([]interface{}) {
+			sRaw := raw.(map[string]interface{})["content_type"]
+			contentTypes = append(contentTypes, aws.String(sRaw.(string)))
+		}
+
+		c.CsvContentTypes = contentTypes
+	}
+
+	if v, ok := m["json_content_types"]; ok && len(v.([]interface{})) > 0 {
+		contentTypes := make([]*string, 0, len(v.([]interface{})))
+
+		for _, raw := range v.([]interface{}) {
+			sRaw := raw.(map[string]interface{})["content_type"]
+			contentTypes = append(contentTypes, aws.String(sRaw.(string)))
+		}
+
+		c.JsonContentTypes = contentTypes
+	}
+
+	return c
+}
+
+func flattenSagemakerCaptureContentTypeHeader(contentTypeHeader *sagemaker.CaptureContentTypeHeader) []map[string]interface{} {
+	if contentTypeHeader == nil {
+		return []map[string]interface{}{}
+	}
+
+	l := make(map[string]interface{})
+
+	if contentTypeHeader.CsvContentTypes != nil {
+		csvTypes := make([]map[string]interface{}, 0, len(contentTypeHeader.CsvContentTypes))
+
+		for _, csvType := range contentTypeHeader.CsvContentTypes {
+			m := map[string]interface{}{
+				"content_type": aws.StringValue(csvType),
+			}
+			csvTypes = append(csvTypes, m)
+		}
+
+		l["csv_content_types"] = csvTypes
+	}
```

Comment:
again as above, this will reduced to:

`l["csv_content_types"] = flattenStringSet(contentTypeHeader.CsvContentTypes)`